### PR TITLE
chore:  移除所有 @deprecated 标记 api

### LINKED ISSUE #2016: 🙏  移除所有 @deprecated api 和 相关代码
### 🖋 Description

- next 分支 移除 废弃的 api 和 兼容代码
- master 分支增加相应的 `@deprecated` 提示 eg.
image

### 🧐 Suggest the API

| name | type | default | description |
| ---- | ---- | ------- | ----------- |
| -    | -    | -       | -           |

### PR DESCRIPTION
**Pull request by wjgogogo**
🔧 Chore

- [x] 移除所有 @deprecated 标记 api, close #2016 
- [x] 文档更新

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -327,7 +327,7 @@ importers:
       '@antv/dumi-theme-antv': 0.3.5_dumi@2.1.1
       '@antv/g-canvas': 0.5.13
       '@antv/s2': link:../packages/s2-core
-      '@antv/s2-react': link:../packages/s2-react
+      '@antv/s2-react': 2.0.0-next.2_ekq5dgpvi24jylo2hahsfpc3h4
```
Comment (lcx-seima) [inline pnpm-lock.yaml:330]: 这个为啥变了

**Review by lijinke666 — APPROVED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (lijinke666) [reply pnpm-lock.yaml:330]: image

**Pull request merged**